spec/ddtrace/contrib/extensions_spec.rb

```diff
@@ -71,7 +68,28 @@
     describe Datadog::Configuration::Settings do
       include_context 'registry with integration'
 
-      subject(:settings) { described_class.new(registry: registry) }
+      subject(:settings) { described_class.new }
+
+      before { stub_const('Datadog::Contrib::REGISTRY', registry) }
+
+      describe '.registry' do
+        it 'return global REGISTRY on deprecated access' do
+          expect(Datadog.logger).to receive(:warn).with(/Deprecated access to `Datadog.configuration.registry`/)
+
+          expect(settings.registry).to be(Datadog::Contrib::REGISTRY)
+        end
+      end
+
+      describe '.registry=' do
+        it 'to not change registry on deprecated assignment attempt' do
+          expect(Datadog.logger).to receive(:warn).with(/no longer supported and was ignored/)
+
+          settings.registry = double
+
+          allow(Datadog.logger).to receive(:warn)
```

Comment:
Minor: Is this `allow` still needed, seeing as there's an `expect` on the logger above?

Reply:
It is, because `settings.registry` on the following line emits another warning, thus making the exception trip twice (and fail).